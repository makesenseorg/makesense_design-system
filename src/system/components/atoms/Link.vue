<template>
  <component :is="component">
    <slot></slot>
  </component>
</template>
<script>
export default {
  name: "MksLink",
  computed: {
    component() {
      // todo : return nuxt link or router link depending on app type
      // todo : detect app type
      // todo : detect starts with https
      return "router-link" || "nuxt-link" || "a";
    }
  }
};
</script>
