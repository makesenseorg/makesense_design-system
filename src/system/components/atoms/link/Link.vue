<template>
  <component
    :is="tag"
    :to="!external ? to : undefined"
    :href="external ? to : undefined"
    :target="external ? '_blank' : undefined"
    :class="{ link: true, 'link--menu': type === 'menu' }"
    :title="
      `${title || ''}${external ? ` (${$MKSlocale['externalLink']})` : ''}`
    "
  >
    <!-- @slot Text of the link-->
    <slot></slot>
  </component>
</template>
<script>
/**
 * Link to internal pages or external pages
 * @version 0.2.0
 */

export default {
  name: "MksLink",
  props: {
    to: {
      type: [String, Object],
      required: true
    },
    /** For usage in the Navigation, adds styling when active link. */
    type: {
      type: String
    },
    /** For accessibility purposes */
    title: {
      type: String
    }
  },
  computed: {
    external() {
      return (
        (typeof this.to === "string" &&
          (this.to.startsWith("http") ||
            this.to.startsWith("//") ||
            this.to.startsWith("www"))) ||
        this.to.startsWith("mailto:")
      );
    },
    tag() {
      if (this.external) {
        return "a";
      }
      if (!this.$nuxt) {
        return "router-link";
      }
      return "nuxt-link";
    }
  }
};
</script>
<style lang="scss" scoped>
.link {
  font-size: inherit;
  font-weight: inherit;
  cursor: pointer;
  text-decoration: none;
  color: inherit;
  border-bottom: $border-width-m solid;
  border-radius: $border-radius-full;

  &.exact-active,
  &.router-link-exact-active,
  &.nuxt-link-exact-active {
    color: $color-primary;
    border-color: $color-primary;
  }

  &:focus {
    @include focus($color-primary);
  }

  &--menu {
    font-size: inherit;
    font-weight: inherit;
    border-bottom: none;

    &:hover {
      opacity: $opacity-hover;
    }

    // &.exact-active,
    // &.router-link-exact-active,
    // &.nuxt-link-exact-active {
    //   font-size: inherit;
    //   color: $color-secondary;
    //   position: relative;

    //   &:focus {
    //     @include focus($color-secondary);
    //   }
    // }
  }
}
</style>
<docs>

## External links 

The component automatically detects an external link if it starts with <code>http</code>, <code>//</code> or <code>www</code>.

If you add a <code>title</code> attribute, it will be inherited.

```jsx

<mks-link to="https://google.com" title="Search for something">An external link</mks-link>

```


## Internal links

If it is an internal link, it automatically detects if the app is Vue or Nuxt and uses either <code>&lt;router-link&gt; </code> or <code>&lt;nuxt-link&gt;</code>.

```jsx
<mks-link to="/design-tokens">An internal link with string</mks-link>
<br><br>
<mks-link v-bind:to="{ name: 'Usage'}">An internal link with object</mks-link>
```

## Active page

If the link is internal and refers to the current page, styling is applied.

```jsx
<mks-link to="/atoms/mkslink">An active link</mks-link>
<mks-link to="/atoms/mkslink" type="menu">An active menu link</mks-link>
```
</docs>
