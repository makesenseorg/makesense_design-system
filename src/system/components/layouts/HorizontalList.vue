<template>
  <component :is="tag" class="horizontal-list">
    <slot></slot>
  </component>
</template>
<script>
/**
 * Displays items in a horizontal line
 * @version 0.2.0
 */
export default {
  name: "MksHorizontalList",
  release: "0.2.0",
  props: {
    tag: {
      type: String,
      default: "div"
    }
  }
};
</script>
<style lang="scss" scoped>
.horizontal-list {
  @include flex;
  align-items: center;

  & > * {
    list-style-type: none;
    margin-right: $space-base;

    &:last-child {
      margin-right: 0;
    }
  }
}
</style>
<docs>

```jsx
<mks-horizontal-list>
  <mks-tag>Tag 1</mks-tag>
  <mks-tag>Tag 2</mks-tag>
  <mks-tag>Tag 3</mks-tag>
</mks-horizontal-list>

```


</docs>
