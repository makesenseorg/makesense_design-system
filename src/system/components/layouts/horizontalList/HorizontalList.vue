<template>
  <component
    :is="tag"
    :class="`horizontal-list horizontal-list--spacing-${spacing}`"
  >
    <slot></slot>
  </component>
</template>
<script>
/**
 * Deprecated: Use Spacer instead - Displays items in a horizontal line
 * @version 0.2.0
 * @deprecated
 */
export default {
  name: "MksHorizontalList",
  status: "deprecated",
  release: "0.2.0",
  props: {
    tag: {
      type: String,
      default: "div"
    },
    spacing: {
      type: String,
      default: "default",
      validator: function(value) {
        return ["default", "none", "small", "large"].indexOf(value) !== -1;
      }
    }
  }
};
</script>
<style lang="scss" scoped>
.horizontal-list {
  @include flex;
  align-items: center;

  &--spacing-small > * {
    margin-right: $space-xs;
  }
  &--spacing-default > * {
    margin-right: $space-m;
  }
  &--spacing-large > * {
    margin-right: $space-l;
  }

  & > * {
    list-style-type: none;

    &:last-child {
      margin-right: 0;
    }
  }
}
</style>
<docs>

Deprecated: Use <mks-link to="/layouts/mksspacer">Spacer</mks-link> instead

```jsx
<mks-horizontal-list spacing="none">
  <mks-tag>Tag 1</mks-tag>
  <mks-tag>Tag 2</mks-tag>
  <mks-tag>Tag 3</mks-tag>
</mks-horizontal-list>

<mks-horizontal-list spacing="small">
  <mks-tag>Tag 1</mks-tag>
  <mks-tag>Tag 2</mks-tag>
  <mks-tag>Tag 3</mks-tag>
</mks-horizontal-list>

<mks-horizontal-list>
  <mks-tag>Tag 1</mks-tag>
  <mks-tag>Tag 2</mks-tag>
  <mks-tag>Tag 3</mks-tag>
</mks-horizontal-list>

<mks-horizontal-list spacing="large">
  <mks-tag>Tag 1</mks-tag>
  <mks-tag>Tag 2</mks-tag>
  <mks-tag>Tag 3</mks-tag>
</mks-horizontal-list>

```


</docs>
