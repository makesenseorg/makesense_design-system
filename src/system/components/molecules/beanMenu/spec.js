import { shallowMount, createLocalVue } from "@vue/test-utils";
import Comp from "./BeanMenu.vue";
import HorizontalList from "../../layouts/horizontalList/HorizontalList.vue";
import Link from "../../atoms/link/Link.vue";

const localVue = createLocalVue();
localVue.component("MksHorizontalList", HorizontalList);
localVue.component("MksLink", Link);

describe("BeanMenu.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      localVue,
      propsData: {
        links: [
          {
            label: "link 1",
            to: "https://example.com",
          },
          {
            label: "link 2",
            to: "https://example2.com",
          },
        ],
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
