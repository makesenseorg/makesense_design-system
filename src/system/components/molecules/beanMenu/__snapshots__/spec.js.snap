// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeanMenu.vue matches snapshot 1`] = `
<nav
  class="bean-menu"
>
  <mks-horizontal-list-stub
    class="bean-menu__list"
    spacing="default"
    tag="ul"
  >
    <li
      class="bean-menu__item"
    >
      <mks-link-stub
        to="https://example.com"
        type="menu"
      >
        link 1
      </mks-link-stub>
    </li>
    <li
      class="bean-menu__item"
    >
      <mks-link-stub
        to="https://example2.com"
        type="menu"
      >
        link 2
      </mks-link-stub>
    </li>
  </mks-horizontal-list-stub>
</nav>
`;
